mission/src/views/ItemInfo.vue

```diff
@@ -40,17 +40,22 @@
 
     <footer class="item-info-footer" data-test="footer-wrapper">
       <button
-        class="w3-round-large w3-black w3-xlarge w3-padding-small"
+        class="
+          w3-orange w3-xlarge w3-round-large w3-padding-small w3-text-white
+        "
         data-test="footer-button"
-        @click="showModal"
+        @click="showModal(), postCartItem(item)"
```

Comment:
두 메서드 내의 기능을 하나의 메서드에 합쳐서 call하는 것이 더 보기 좋을 것 같은데, 혹시 나누신 이유가 있으실까요?? + 아니면 showModal 메서드 안 맨마지막 줄에 조건을 걸어서 postCartItem 메서드를 호출해도 괜찮을 것 같다고 생각했습니다!

Reply:
안녕하세요, 민선님! 일단은 두 함수를 등록할 때 일부러 나눈 건 아니었어요 😃
실제로 '장바구니에 담기' 기능을 구현하게 될 줄 모르고, 예전에 Modal 컴포넌트와 함께 `showModal`이라는 함수를 구현해뒀는데요-

`showModal`이 Modal을 열어주는 역할의 지역 함수라면, `postCartItem`은 장바구니에 상품을 담아주는 역할의 전역 함수잖아요?
그래서 각기 다른 역할의 두 함수를 굳이 합치지 않고, 하나의 이벤트가 작동할 때에 두 개의 함수가 실행되게끔 구현한 거였어요 😯

Reply:
제안해주신 내용에 대해서 조금 더 고민해보자면,

- 우선 `showModal`이라는 이름 아래에 `postCartItem`을 넣어주는 건 의미적인 차원에서 아닌 것 같고,
- 오히려 `showModal`이라는 이름을 바꾸고, 두 함수의 역할을 하나로 합쳐주는 방법은 적극 고려해볼 수 있을 것 같아요!!
_(제가 함수의 역할이 명확히 구분되고, 그 역할에 맞춰 이름 짓는 걸 되게 중요하게 생각해서요오...)_

Reply:
제안해주신 내용을 적용해보려고 아래와 같이 시도를 해보았어요 😁

일단, 두 함수의 기능을 하나로 합치되, 함수 이름을 `showModal`에서 `postCartItemWithModal`로 바꿔서
`showModal`과 `postCartItem` 두 함수의 기능이 하나로 합쳐져 있다는 사실을 함수의 이름을 통해서 드러내려고 노력했어요!

```
postCartItemWithModal() {
  this.postCartItem(this.item);
  this.isModalShown = true;
},
```

그리고 아래와 같이 메서드 바인딩 방식을 바꿔줬는데요, 막상 민선님 말씀처럼 수정해보니까 훨씬 깔끔하고 좋네요 👏👏👏

```suggestion
        @click="postCartItemWithModal"
```

Reply:
@jambottle 이렇게 자세하게 써주시다니 👍 👍 두 메서드의 역할이 확실하게 다른건 있네요!!
제 의견 반영해서 수정해주신 것 넘넘 감사합니다 😄 저도 배웠어요

Reply:
아뇨, 민선님이 질문 주신 덕분에 한 번 더 생각해볼 계기가 생겨서 좋았습니다 🙏